thresholdkey_generator.go

```diff
@@ -58,7 +59,7 @@ func GetThresholdKeyGenerator(nbits, TotalNumberOfDecryptionServers, Threshold i
 }
 
 func (tkg *ThresholdKeyGenerator) generateSafePrimes() (*big.Int, *big.Int, error) {
-	return GenerateSafePrimes(tkg.nbits, tkg.Random)
+	return GenerateSafePrime(tkg.nbits, 4, 120*time.Second, tkg.Random)
```

Comment:
What do you think about extracting the concurrency level value `4` to a constant on the file level?